cogip/models/parameter.py

```diff
@@ -0,0 +1,316 @@
+            ParameterValidationFailed: If parameter validation failed on the embedded side
+            ParameterReadOnly: If parameter is read-only on the embedded side
+            ParameterNotFound: If parameter not found on the embedded side
+        """
+
+        # Verify that the name matches
+        if message.key_hash != hash(self):
+            raise ValueError(f"Key hash mismatch: expected '{hash(self)}', got '{message.key_hash}'")
+
+        if isinstance(message, PB_ParameterSetResponse):
+            # Check status and raise appropriate exceptions
+            match message.status:
+                case PB_ParameterStatus.VALIDATION_FAILED:
+                    raise ParameterValidationFailed(f"Parameter '{self.name}' validation failed: value out of bounds")
+                case PB_ParameterStatus.READ_ONLY:
+                    raise ParameterReadOnly(f"Parameter '{self.name}' is read-only and cannot be modified")
+                case PB_ParameterStatus.NOT_FOUND:
+                    raise ParameterNotFound(f"Parameter '{self.name}' not found in registry")
+                case PB_ParameterStatus.SUCCESS:
+                    pass  # Operation succeeded, nothing to do
+        elif isinstance(message, PB_ParameterGetResponse):
+            # Get the name of the field defined in the oneof
+            which_field = message.value.WhichOneof("value")
+
+            if which_field is None:
+                raise ValueError("No value set in ParameterGetResponse, parameter not found")
+
+            # Get the value of the active field
+            content = getattr(message.value, which_field)
+
+            # Update the parameter content
+            self.value.content = content
+
+
+class ParametersGroup(BaseModel):
+    """Container for a group of parameters with name-based access.
+
+    This class manages a collection of parameters and provides convenient
+    get/set methods using parameter names.
+
+    Attributes:
+        parameters: List of Parameter objects
+    """
+
+    parameters: list[Parameter] = Field(default_factory=list)
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        # Build index for fast lookup by name
+        self._rebuild_index()
+
+    def _rebuild_index(self) -> None:
+        """Rebuild the internal index for parameter name lookup."""
+        self._index: dict[str, int] = {param.name: idx for idx, param in enumerate(self.parameters)}
+
+    def get(self, name: str) -> Parameter:
+        """Get a parameter by its name.
+
+        Args:
+            name: The parameter name
+
+        Returns:
+            The Parameter object
+
+        Raises:
+            KeyError: If the parameter name is not found
+        """
+        if name not in self._index:
+            raise KeyError(f"Parameter '{name}' not found")
+        return self.parameters[self._index[name]]
+
+    def get_value(self, name: str) -> float | int | bool:
```

Comment:
Since you are defining attributes like `__contains__`, `__len__` and `__iter__`, you could also implement `__setitem__` and `__getitem__` instead of `get_value`/`set_value`.

Reply:
Done. get_value/set_value removed.